### PR TITLE
Add release workflow and helper tools

### PR DESCRIPTION
**Pull request by Benedikt-Brunner**
### Summary

Add a release workflow to the project and update the ReadMe.

### CODE REVIEW

File: tools/prepare_release.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    ## update the uv.lock file
+    os.system("uv sync")
+
+    ## commit the changes
+    os.system("git add pyproject.toml")
+    os.system("git add uv.lock")
+    os.system(f'git commit -m "Bump version to {new_version}"')
+    # os.system(f"git push origin {branch_name}")
+
```
Comment (Kobsel) [inline tools/prepare_release.py:62]: Is the push command still intended to be commented out?

**Review by Kobsel — APPROVED** [review]
Looks good. But shouldn't there be a mechanism to prevent the release.py being run before the prepare_release.py?

**Review by Kobsel — CHANGES_REQUESTED** [review]
Looks good. But shouldn't there be a mechanism to prevent the release.py being run before the prepare_release.py?

**Comment by Benedikt-Brunner** [comment]
> Looks good. But shouldn't there be a mechanism to prevent the release.py being run before the prepare_release.py?

Well the two scripts don't really depend on each other. They just represent two different actions, it is the responsibility of the developer to use them correctly. This limitation stems from us not allowing a direct push to main.

**Review by Kobsel — APPROVED** [review]

**Pull request merged**